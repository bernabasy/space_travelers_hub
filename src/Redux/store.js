import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from './Missions/missions';
import dragonSlice from './Bragons/bragonSlice';
import rocketReducer from './Rockets/Rockets';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    missions: missionsReducer,
    dragons: dragonSlice,
  },
});

export default store;
