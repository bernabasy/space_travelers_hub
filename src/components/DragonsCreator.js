/* eslint-disable react/prop-types */
import React from 'react';
import '../styles/dragonStyle.css';

function DragonsCreator(props) {
  const { dragons } = props;
  const {
    id, dragonName, description, flickrImages,
  } = dragons;

  return (
    <div id={id} className="a-dragon">
      <div className="dragons-img-cont">
        <img src={flickrImages[0]} alt="dragon" className="dragons-img" />
      </div>
      <div>
        <h2>{dragonName}</h2>
        <p>
          {' '}
          {description}
          {' '}
        </p>
        <button type="button" className="btn" id="btn">
          Reserve Dragon
        </button>
      </div>
    </div>
  );
}

export default DragonsCreator;

// DragonsCreator.propTypes = {
//   id: propTypes.string.isRequired,
//   dragonName: propTypes.string.isRequired,
//   description: propTypes.string.isRequired,
//   flickrImages: propTypes.string.isRequired,
//   dragons: propTypes.string.isRequired,
// };
