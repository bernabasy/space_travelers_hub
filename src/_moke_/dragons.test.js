import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../Redux/store';
import Dragon from '../pages/Dragons';
import DragonsCreator from '../components/DragonsCreator';
import DragonTest from './dragonReducer';

const dragons = {
  dragonName: 'Heloo',
  description: 'Heloof',
  flickrImages: 'Helookn',
  reserved: false,
  id: 2,
};

jest.mock('axios');
describe('dragon must pass the test', () => {
  test('added dragon must return data', () => {
    expect(DragonTest.addedRockets()).toBeDefined();
  });
  test('added dragon return value length must be 3', () => {
    expect(DragonTest.addedRockets()).toHaveLength(2);
  });
  test('added dragon must return name t', () => {
    expect(DragonTest.addedRockets()[1].description).toBe(
      'Never give up on your dream',
    );
  });
});

describe('Dragon component', () => {
  it('renders dragon component', () => {
    const r = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Dragon />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(r).toMatchSnapshot();
  });
  it('renders dragon component', () => {
    const m = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <DragonsCreator dragons={dragons} />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(m).toMatchSnapshot();
  });
});
