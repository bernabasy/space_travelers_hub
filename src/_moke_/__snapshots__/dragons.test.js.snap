// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dragon component renders dragon component 1`] = `
<div
  className="all-dragons"
/>
`;

exports[`Dragon component renders dragon component 2`] = `
<div
  className="all-dragons"
/>
`;
